pallets/liquid-staking/src/types.rs

```diff
@@ -91,7 +91,27 @@ pub struct StakingBondCall<T: Config> {
 pub struct StakingBondExtraCall<T: Config> {
```

Comment:
There are too many constraints once using`<T: Config>`.  `<Balance>` is enough.  https://github.com/parallel-finance/parallel/blob/master/pallets/liquid-staking-v2/src/types.rs#L14

Reply:
Balance I think may requires some trait bounds to make sure it's a `number`

Reply:
or codec

Reply:
When you bond it with BalanceOf<T>, the Balance will be derived as `MultiCurrency<<T as frame_system::Config>::AccountId>>::Balance` which is u128.

https://github.com/parallel-finance/parallel/blob/master/pallets/liquid-staking-v2/src/lib.rs#L177

Reply:
but what's the value of this change? `Balance` is even longer than `Config`

Reply:
using https://github.com/parallel-finance/parallel/blob/master/pallets/liquid-staking-v2/src/types.rs#L14 I'll need to add trait bound

Reply:
You needn't. 
The type Balance in MultiCurrency like this:
`type Balance: AtLeast32BitUnsigned + FullCodec + Copy + MaybeSerializeDeserialize + Debug + Default;`

AtLeast32BitUnsigned contains BaseArithmetic which already have the trait Bounded.
You can treat as number by using BaseArithmetic trait.

Reply:
maybe you can write some code here so that  I can better understand

```rust
/// Relaychain staking.unbond call arguments
#[derive(Clone, Encode, Decode, RuntimeDebug)]
pub struct StakingRebondCall<T: Config> {
    /// [pallet index, call index]
    pub call_index: [u8; 2],
    /// Rebond amount
    #[codec(compact)]
    pub value: T::Balance,
}
```
do you want to do this?

Reply:
```suggestion
/// Relaychain staking.unbond call arguments
#[derive(Clone, Encode, Decode, RuntimeDebug)]
pub struct StakingRebondCall<Balance> {
    /// [pallet index, call index]
    pub call_index: [u8; 2],
    /// Rebond amount
    #[codec(compact)]
    pub value: Balance,
}
```